src/main/java/com/serenitydojo/Catalog.java

```diff
@@ -3,18 +3,97 @@
+    Map<Fruit, Double> fruitToUnitPrice = new HashMap<Fruit, Double>();
+    List<CatalogItem> availableFruits = new ArrayList<CatalogItem>();
+    List<Fruit> fruitsEligibleForDiscount = new ArrayList<Fruit>();
+
+    public Catalog(){
+        fruitToUnitPrice.put(Fruit.Apple, 4.00);
+        fruitToUnitPrice.put(Fruit.Orange, 5.50);
+        fruitToUnitPrice.put(Fruit.Banana, 6.00);
+        fruitToUnitPrice.put(Fruit.Pear, 4.50);
+        fruitToUnitPrice.put(Fruit.Grapefruit, 7.00);
     }
 
     public static Catalog withItems(CatalogItem... catalogItems) {
-        throw new RuntimeException("TODO, create catalog and add items to the list of available fruits");
-        // Catalog catalog = new Catalog();
-        // for (CatalogItem catalogItem : catalogItems) {
-        //     catalog.availableFruits.add(catalogItem);
-        // }
-        // return catalog
+        //throw new RuntimeException("TODO, create catalog and add items to the list of available fruits"); - Done
+        Catalog catalog = new Catalog();
+        for (CatalogItem catalogItem : catalogItems) {
+            catalog.availableFruits.add(catalogItem);
+        }
+        return catalog;
+    }
+
+    public List<CatalogItem> getAvailableFruits(){
+        //Ash to reviewer: Which one's preferred? Mutate the availableFruits List as I did here OR sort on streams instead and produce a new sorted List
+        availableFruits.sort(Comparator.comparing(catalogItem -> catalogItem.getFruit().name()));
+        return availableFruits;
+
+        //        Alternate solution: sort on streams instead and produce a new sorted List
+        //        List<CatalogItem> availableFruitsSorted =
+//                availableFruits.stream()
+//                .sorted(Comparator.comparing(catalogItem -> catalogItem.getFruit()))
+//                .collect(Collectors.toList());
+//
+//       return availableFruitsSorted;
+//        return availableFruits;
+    }
+    public void setUnitPriceOf(Fruit fruit, double unitPrice) {
+        //Ash: Commenting the exception now that I've created the map
+        //throw new RuntimeException("TODO, create a map to keep track of fruits and their prices");
+
+        // Ash to reviewer: To get the following preexisting line of code to work, I'd have to change the data type of the Map's key from Fruit to String
+        // fruitToPrice.put(fruit.name(), unitPrice);
+
+        fruitToUnitPrice.put(fruit, unitPrice);
+    }
+
+    public Double getUnitPriceOf(Fruit fruit) throws FruitUnavailableException{
+        Boolean isFruitAvailable = false;
+
+        // Ash to reviewer: Not sure in response to what exactly we wanted the "FruitUnavailableException", so I included it here too
+        checkFruitAvailability(fruit);
+        return fruitToUnitPrice.get(fruit);
+    }
+
+    public Double calculateFruitPriceForQuantity(Fruit fruit, Integer quantityInKilos) throws FruitUnavailableException{
+        Double price = getUnitPriceOf(fruit);
```

Comment:
```suggestion
        long priceInCents = getUnitPriceOf(fruit);
```
you don't need to change that here... but in general you don't want to use decimal numbers to deal with money. decimal numbers can't accurately store specific numbers and then you get compounding rounding errors and suddenly people lose millions :) just fyi 
there is still the problem of "so what happens if the discount results in 100099.25 cents" :shrug: :)

Reply:
@globalworming great insight! What's a better way to store prices when the ask is a particularly $5.50, for example? Store it as $5.5 as a Long?

Reply:
currencies (always?) have a non-decimal smallest measures, like for dollar it's cent. instead of `double dollars = 5.5` you should use `long cent = 550` to prevent these errors with decimal accuracy.

Reply:
😮 that's enlightening. I declare this insight as "TIL" of the day😸